rust/arrow/src/util/display.rs

```diff
@@ -44,6 +44,66 @@ macro_rules! make_string {
     }};
 }
 
+macro_rules! make_string_interval_year_month {
+    ($column: ident, $row: ident) => {{
+        let array = $column
+            .as_any()
+            .downcast_ref::<array::IntervalYearMonthArray>()
+            .unwrap();
+
+        let s = if array.is_null($row) {
+            "0 years 0 mons 0 days 0 hours 0 mins 0.00 secs".to_string()
+        } else {
+            let interval = array.value($row) as f64;
+            let years = (interval / 12_f64).floor();
+            let month = interval - (years * 12_f64);
+
+            format!(
+                "{} years {} mons 0 days 0 hours 0 mins 0.00 secs",
+                years, month,
+            )
+        };
+
+        Ok(s)
+    }};
+}
+
+macro_rules! make_string_interval_day_time {
+    ($column: ident, $row: ident) => {{
+        let array = $column
+            .as_any()
+            .downcast_ref::<array::IntervalDayTimeArray>()
+            .unwrap();
+
+        let s = if array.is_null($row) {
+            "0 years 0 mons 0 days 0 hours 0 mins 0.00 secs".to_string()
```

Comment:
I wonder if `""` would be a more appropriate representation for `NULL`, to conform with the handling of null strings:

https://github.com/apache/arrow/pull/9373/files#diff-821be3a8a9239cdd12ab2b3c979e174f7b936aa6bfad4b9bc76489b9923cf747R164

I also think `"0 years 0 mons 0 days 0 hours 0 mins 0.00 secs"` might be the same representation if `value = 0` (so you couldn't distinguish between `0` and `NULL`(

Reply:
fixed.